### DISCUSSION TITLE
Next.js - Zustand Example - getServerSideProps

### DISCUSSION THREAD

**Discussion by glitch-txs** [vercel/next.js#47330] [Help] [answered]
### Summary

Hi @Mojtaba-NA or anyone, I'm trying to follow the [with-zustand](https://github.com/vercel/next.js/tree/canary/examples/with-zustand) example for SSR. I wanted to ask if it's necessary to use getServerSideProps as I would like to avoid it otherwise I would need to instantiate the store on every page of the website. So would this still work if I remove that and I just instantiate the store on _app as it currently is? Thanks

**Comment by Mojtaba-NA** [comment]
Hi. Instantiate the store on _app with what data? if you don't provide it with data, it will default to default store value. Is that okay?
Can you explain what kind of store do you want?

**Comment by glitch-txs** [comment]
Hi @Mojtaba-NA , thanks a lot for helpinh.

I'm trying to create a client side store using SSR/Next.js . There are some default values in the store and once it gets into the client some eventListeners will update them and some should just keep its default value.

would something [like this](https://github.com/glitch-txs/zustand-nextjs/blob/main/official-next-example/src/pages/_app.tsx) work? Thanks

**Reply by Mojtaba-NA** [reply]
@glitch-txs If it's client side only, you don't need the StoreProvider, just create a store and export it and use it.
see [https://github.com/pmndrs/zustand#first-create-a-store](https://github.com/pmndrs/zustand#first-create-a-store)

**Reply by glitch-txs** [reply]
I was told this might cause memory leaks on global states. As the SSR framework will try to run the store on server. Please take a look at this issue:

https://github.com/pmndrs/zustand/issues/182

**Comment by Mojtaba-NA** [comment, marked as answer]
I understand the issue and I can reproduce it but I don't understand why would they use the store like that (set the value on the server without persisting it in cookies).
if you don't use it like that, you wouldn't run into that issue. (just set the store on the client)

if you need to, you can persist the store in sessionStorage or localStorage (or even cookies).

persist middleware (easily used with session or local which would make it client side only)
[https://github.com/pmndrs/zustand/blob/main/docs/integrations/persisting-store-data.md](https://github.com/pmndrs/zustand/blob/main/docs/integrations/persisting-store-data.md)

if you want the store to persist in cookie, this is what I've came up with (which would require getServerSideProps):
[https://github.com/Mojtaba-NA/universal-zustand](https://github.com/Mojtaba-NA/universal-zustand)

**Reply by glitch-txs** [reply]
Thanks a lot for the insight 🙂
